pkg/resources/server_test.go

```diff
@@ -113,24 +119,56 @@ var _ = Describe("Server", func() {
 				err := rs.Start()
 				Expect(err).NotTo(HaveOccurred())
 
+				// channel should receive after calling rs.restart()
+				go Eventually(rs.termSignal, time.Second*5).Should(Receive())
```

Comment:
I don't need to run the Eventually on separate thread. This may lead to some other complication. 
Add `Eventually(rs.termSignal, time.Second*5).Should(Receive())` after `err = rs.restart()`

Reply:
done